foam_internalwrapper/tests/test_internalwrapper.py

```diff
@@ -0,0 +1,275 @@
+
+        for cell in self.mesh.iter_cells():
+            cell_w = mesh_inside_wrapper.get_cell(cell.uid)
+            self.assertEqual(cell.uid, cell_w.uid)
+            self.assertEqual(set(cell.points), set(cell_w.points))
+
+    def test_iter_meshes(self):
+        """Test iter_meshes method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        wrapper.add_mesh(self.mesh)
+        mesh2 = self.mesh
+        mesh2.name = "mesh2"
+        wrapper.add_mesh(mesh2)
+
+        self.assertEqual(sum(1 for _ in wrapper.iter_meshes()), 2)
+
+    def test_multiple_meshes(self):
+        """Test multiple meshes inside wrapper
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        wrapper.add_mesh(self.mesh)
+        mesh2 = self.mesh
+        mesh2.name = "mesh2"
+        wrapper.add_mesh(mesh2)
+        mesh_inside_wrapper1 = wrapper.get_mesh(self.mesh.name)
+        mesh_inside_wrapper2 = wrapper.get_mesh(mesh2.name)
+
+        self.assertEqual(
+            sum(1 for _ in mesh_inside_wrapper1.iter_points()),
+            sum(1 for _ in mesh_inside_wrapper2.iter_points()))
+
+    def test_add_particles(self):
+        """Test add_particles method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.add_particles(DataContainer())
+
+    def test_get_particles(self):
+        """Test get_particles method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.get_particles('')
+
+    def test_delete_particles(self):
+        """Test delete_particles method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.delete_particles('')
+
+    def test_iter_particles(self):
+        """Test iter_particles method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.iter_particles()
+
+    def test_add_lattice(self):
+        """Test add_lattice method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.add_lattice('')
+
+    def test_get_lattice(self):
+        """Test get_lattice method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.get_lattice('')
+
+    def test_delete_lattice(self):
+        """Test delete_lattice method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.delete_lattice('')
+
+    def test_iter_lattices(self):
+        """Test iter_lattices method
+
+        """
+
+        wrapper = FoamInternalWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.iter_lattices()
+
+    def test_run_time(self):
```

Comment:
> I would keep these separated. There is for example tearDown for files created on IO wrapper side, which is missing on internal wrapper side. If there is a way to make this dependent on wrapper we could do that, but i think that a separate issue should be opened for that. Could you do that @jmarcelogimenez.

@jmarcelogimenez  please open up a issue on GitHub to track this unresolved comment (e.g. "create common testing framework for both test_internalwrapper.py and tests/test_controlwrapper.py")